build.gradle.kts

```diff
@@ -0,0 +1,71 @@
+import com.example.buildsrc.MyExtension
+
+plugins {
+    id("java")
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnitPlatform()
+}
+
+allprojects{
+    extensions.create("Extension", MyExtension::class.java)
+
+    val addElementNoProjectAllScope by extra { _: Unit ->
+        configure<MyExtension> {
+            contents.add("addElementNoProjectAllScope")
+        }
+    }
+
+    //passing a no-argument lambda makes the val initialized with lambda result
+    val addElementFakeLambdaAllScope by extra {
+        configure<MyExtension> {
+            contents.add("addElementFakeLambdaAllScope")
+        }
+    }
+}
+
+val addElementNoProjectExtScope by extra { _: Unit ->
+    configure<MyExtension> {
+        contents.add("addElementNoProjectExtScope")
+    }
+}
+
+//passing a no-argument lambda makes the val initialized with lambda result
+val addElementFakeLambdaExtScope by extra {
+    configure<MyExtension> {
+        contents.add("addElementFakeLambdaExtScope")
+    }
+}
+
+val addElement by extra { project: Project ->
+    project.configure<MyExtension> {
+        contents.add("addElement")
+    }
+}
+
+val printContents by extra (fun(project: Project){
```

Comment:
Is it equivalent to `project: Project -> ...`? or does this form exhibit yet another behavior? 😅

Reply:
It is, and the behaviour is the same, but for example you can NOT specify the return type with the kotlin's lambda syntax as stated here: [SOF](https://stackoverflow.com/a/56311548)